src/components/checkbox/checkbox.jsx

```diff
@@ -53,6 +53,14 @@ const Checkbox = forwardRef(function Checkbox(props, forwardedRef) {
     props.onBlur(event);
   }
 
+  function onClick(event) {
+    if (props.readOnly) event.preventDefault();
+  }
+
+  function onKeyDown(event) {
+    if (props.readOnly && event.key === ' ') event.preventDefault();
+  }
```

Comment:
d'oh.

I'm a bit surprised about this if I'm honest, I guess I would have expected the `readonly` on the native `input` to do this already, but it's my bad for missing the test coverage. Good catch!

Reply:
I looked it up.... HTML spec defines read-only to only stop modifying the `value` of the element. A checkbox does not modify its value. It modifies its checked state. So native read-only functionality does not apply to checkboxes. 🤦 

Random internet answers suggest having hidden checkboxes and a disabled checkbox. It feels like the wrong solution though? 

I went with this solution in the end. Markup is simple. Implementation details are simple too.